src/Browscap/Data/DataCollection.php

```diff
@@ -75,52 +71,39 @@ class DataCollection
     /**
      * @var string[]
      */
-    private $allDivision = [];
+    private $allDivisions = [];
 
     /**
-     * Create a new data collection for the specified version
-     *
-     * @param string $version
+     * @var DivisionFactory
      */
-    public function __construct($version)
-    {
-        $this->version        = $version;
-        $this->generationDate = new \DateTime();
-    }
+    private $divisionFactory;
 
     /**
-     * @param \Psr\Log\LoggerInterface $logger
+     * Create a new data collection for the specified version
      *
-     * @return \Browscap\Data\DataCollection
-     */
-    public function setLogger(LoggerInterface $logger)
-    {
-        $this->logger = $logger;
-
-        return $this;
-    }
-
-    /**
-     * @return \Psr\Log\LoggerInterface $logger
+     * @param string                   $version
+     * @param \Psr\Log\LoggerInterface $logger
```

Comment:
Don't need to use fully qualified class name here (`LoggerInterface` is already imported)